### PR TITLE
[Misc] Add unit tests for min_p Triton sampling kernel

### PR DESCRIPTION
**Pull request by gkuwanto**
## Purpose
 
Add dedicated unit tests for the `min_p` Triton sampling kernel (`vllm/v1/worker/gpu/sample/min_p.py`).
 
Currently, `min_p` filtering is only tested indirectly through e2e sampling parameter tests. This PR adds direct kernel-level tests that validate correctness of the `apply_min_p` function against a pure PyTorch reference implementation.
 
## Test Plan
 
```bash
python -m pytest tests/v1/sample/test_min_p_kernel.py -v
```
 
### Tests added:
 
- `test_min_p_basic` — validates filtering across multiple vocab sizes (128, 1024, 32000) against a reference implementation
- `test_min_p_zero_is_noop` — verifies `min_p=0.0` triggers the early return path and leaves logits unchanged
- `test_min_p_one_keeps_only_max` — verifies `min_p=1.0` masks everything except the argmax
- `test_min_p_all_equal_logits` — edge case where all logits are identical (none should be filtered)
- `test_min_p_shared_request_mapping` — tests `expanded_idx_mapping` indirection with multiple tokens sharing the same request
- `test_min_p_mixed_zero_and_nonzero` — mix of active and inactive min_p values in a single batch
- `test_min_p_already_neg_inf` — logits that are already `-inf` remain `-inf`
- `test_min_p_threshold_correctness` — parametrized exact threshold verification (`logit >= max + log(min_p)`) across 4 min_p values
 
## Test Result
 
See github CI result

---
 
- [x] The purpose of the PR
- [x] The test plan
- [x] The test results
- [ ] ~Documentation update~ (N/A — test-only change)
- [ ] ~Release notes~ (N/A — test-only change)

---
This PR includes AI-assisted code generation. All tests were reviewed, understood, and validated by the author.

Co-authored-by: Claude
Signed-off-by: Garry Kuwanto <gkuwanto@gmail.com>

### CODE REVIEW

File: tests/v1/sample/test_min_p_kernel.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        if logits[0, i].item() >= threshold:
+            assert result[0, i].item() == logits[0, i].item(), (
+                f"Token {i} should have survived (logit={logits[0, i].item()}, "
+                f"threshold={threshold})"
+            )
+        else:
+            assert result[0, i].item() == float("-inf"), (
+                f"Token {i} should have been filtered (logit={logits[0, i].item()}, "
+                f"threshold={threshold})"
+            )
```
[the remark this answers is not in the document]
Comment (gkuwanto) [reply tests/v1/sample/test_min_p_kernel.py:177]: Values like 1e-46 underflow to 0.0 in float32, which is the dtype used throughout vLLM's sampling pipeline. This is expected behavior rather than a bug; min_p values in practice range from ~0.01 to 0.9, well within float32 precision. The kernel correctly matches the behavior of the rest of the sampling path. Adding an xfail test for float32 underflow would test IEEE 754 semantics rather than kernel correctness.
